src/clock-weather-card.ts

```diff
@@ -457,29 +460,31 @@ export class ClockWeatherCard extends LitElement {
       return localize(key, this.getLocale());
   }
 
-  private extractDailyForecasts(forecasts: WeatherForecast[], days: number): MergedWeatherForecast[] {
+  private extractForecasts(forecasts: WeatherForecast[], items: number, hourly: boolean): MergedWeatherForecast[] {
     const agg = forecasts.reduce((forecasts, forecast) => {
-      const day = new Date(forecast.datetime).getDate();
-      forecasts[day] = forecasts[day] || [];
-      forecasts[day].push(forecast);
+      const d = new Date(forecast.datetime);
+      const unit = !hourly ? d.getDate() : d.getHours();
+      forecasts[unit] = forecasts[unit] || [];
```

Comment:
there is an issue on how the aggregation works now. 
forecasts will be aggregated by hours, which means for example a forecast at 2pm for today will be combined with a forecast for 2pm tomorrow and later `calculateAverageForecast` will calculate the average of both.

this causes another problem where a maximum of 24 hours can be displayed, even if the weather provider delivers more than 24 hours

Reply:
Thanks for the heads-up. I have added the date to the aggregation as well to fix this.